pkg/odo/cli/catalog/list/components.go

```diff
@@ -199,12 +199,12 @@ func (o *ListComponentsOptions) printCatalogList(w io.Writer, catalogList []cata
 }
 
 func (o *ListComponentsOptions) printDevfileCatalogList(w io.Writer, catalogDevfileList []catalog.DevfileComponentType, supported string) {
-	for _, devfileComponent := range catalogDevfileList {
+	for i := len(catalogDevfileList) - 1; i >= 0; i-- {
```

Comment:
Do we need to loop backwards here?

I think looping backwards here is what's causing the devfiles in `odo catalog list` to get listed in non-alphabetical order:
```
Johns-MacBook-Pro-3:odo johncollier$ odo catalog list components
Odo Devfile Components:
NAME                 DESCRIPTION                            REGISTRY
nodejs               Stack with NodeJS 12                   DefaultDevfileRegistry
java-springboot      Spring Boot® using Java                DefaultDevfileRegistry
java-quarkus         Upstream Quarkus with Java+GraalVM     DefaultDevfileRegistry
java-openliberty     Open Liberty microservice in Java      DefaultDevfileRegistry
java-maven           Upstream Maven and OpenJDK 11          DefaultDevfileRegistry
```

Reply:
I change the way we store the latest added registry, so with the latest change, the internal order of the components within the registry won't change but the external order (registry order) is preserved.

Reply:
And we you are right, that the issue, the latest change doesn't need to loop backwards.

Reply:
Makes sense 👍